### PR TITLE
feat(panel): Add component tokens

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #7180 

This will also serve as a pattern for consuming component like Dialog, Sheet, etc.

## Summary

`--calcite-panel-corner-radius`: Specifies the corner radius of the component.
`--calcite-panel-heading-text-color`: Specifies the text color of the component's `heading`.
`--calcite-panel-description-text-color`: Specifies the text color of the component's `description`.
`--calcite-panel-border-color`: Specifies the border color of the component.

`--calcite-panel-background-color`: Specifies the background color of the component.
`--calcite-panel-header-background-color`: Specifies the background color of the component's footer.
`--calcite-panel-footer-background-color`: Specifies the background color of the component's footer.

`--calcite-panel-space`: Specifies the padding of the component's `"unnamed (default)"` slot.
`--calcite-panel-header-content-space`: Specifies the padding of the `"header-content"` slot.
`--calcite-panel-footer-space`: Specifies the padding of the component's footer.

`--calcite-action-background-color`: Specifies the background color of the component's `closable`, `collapsible`, and `back` Action components. Will apply to any slotted Calcite Action components.
`--calcite-action-background-color-hover`: Specifies the background color of the component's `closable`, `collapsible`, and `back` Action components when hovered. Will apply to any slotted Calcite Action components.
`--calcite-action-background-color-pressed`: Specifies the background color of the component's `closable`, `collapsible`, and `back` Action components when pressed. Will apply to any slotted Calcite Action components.
`--calcite-action-text-color-hover`: Specifies the text and icon color of the component's `closable`, `collapsible`, and `back` Action components when hovered. Will apply to any slotted Calcite Action components.
`--calcite-action-text-color-pressed`: Specifies the text and icon color of the component's `closable`, `collapsible`, and `back` Action components when pressed. Will apply to any slotted Calcite Action components.
`--calcite-popover-border-color`: Specifies the border color of the component's Calcite Popover component, which is used to display a Calcite Action Menu when slotted Calcite Actions are present in the `header-actions-end` slot. Will apply to any slotted Calcite Popover components.

`--calcite-panel-content-space`: [Deprecated] Specifies the padding of the component's content. Use `--calcite-panel-space` instead.
`--calcite-panel-footer-padding`: [Deprecated] Specifies the padding of the component's footer. Use `--calcite-panel-footer-space` instead.
`--calcite-panel-header-border-block-end`: [Deprecated] Specifies the component header's block end border. Use `--calcite-panel-border-color` instead.

### CODE REVIEW

File: packages/calcite-components/src/components/panel/panel.scss [hunk 1]
```diff
@@ -2,11 +2,30 @@
...
+ * @prop --calcite-action-background-color-hover: Specifies the background color of the component's `closable`, `collapsible`, and `back` Action components when hovered. Will apply to any slotted Calcite Action components.
+ * @prop --calcite-action-background-color-pressed: Specifies the background color of the component's `closable`, `collapsible`, and `back` Action components when pressed. Will apply to any slotted Calcite Action components.
+ * @prop --calcite-action-text-color-hover: Specifies the text and icon color of the component's `closable`, `collapsible`, and `back` Action components when hovered. Will apply to any slotted Calcite Action components.
+ * @prop --calcite-action-text-color-pressed: Specifies the text and icon color of the component's `closable`, `collapsible`, and `back` Action components when pressed. Will apply to any slotted Calcite Action components.
+ * @prop --calcite-popover-border-color: Specifies the border color of the component's internally rendered Popover component, which is rendered within an Action Menu when slotted Actions are present in the `header-actions-end` slot. Will apply to any slotted Calcite Popover components.
  */
 
 :host {
```
Comment (alisonailea) [inline packages/calcite-components/src/components/panel/panel.scss:31]: `border-radius` needs to be added to the :host or the component doesn't look right.

File: packages/calcite-components/src/components/panel/panel.scss [hunk 2]
```diff
@@ -2,11 +2,30 @@
...
+ * @prop --calcite-panel-heading-text-color: Specifies the text color of the component's `heading`.
+ * @prop --calcite-panel-description-text-color: Specifies the text color of the component's `description`.
+ * @prop --calcite-panel-border-color: Specifies the border color of the component.
+
  * @prop --calcite-panel-background-color: Specifies the background color of the component.
+ * @prop --calcite-panel-header-background-color: Specifies the background color of the component's footer.
+ * @prop --calcite-panel-footer-background-color: Specifies the background color of the component's footer.
+
+ * @prop --calcite-panel-space: Specifies the padding of the component's `"unnamed (default)"` slot.
+ * @prop --calcite-panel-header-content-space: Specifies the padding of the `"header-content"` slot.
```
Comment (alisonailea) [inline packages/calcite-components/src/components/panel/panel.scss:16]: this doesn't appear to do anything. Remove?

File: packages/calcite-components/src/components/panel/panel.scss [hunk 3]
```diff
@@ -2,11 +2,30 @@
...
+ * @prop --calcite-panel-footer-space: Specifies the padding of the component's footer.
+
+ * @prop --calcite-panel-content-space: [Deprecated] Specifies the padding of the component's content. Use `--calcite-panel-space` instead.
+ * @prop --calcite-panel-footer-padding: [Deprecated] Specifies the padding of the component's footer. Use `--calcite-panel-footer-space` instead.
+ * @prop --calcite-panel-header-border-block-end: [Deprecated] Specifies the component header's block end border. Use `--calcite-panel-border-color` instead.
```
Comment (alisonailea) [inline packages/calcite-components/src/components/panel/panel.scss:21]: this appears to be overriding `--calcite-panel-border-color`

File: packages/calcite-components/src/components/panel/panel.scss [hunk 4]
```diff
@@ -2,11 +2,30 @@
...
+
  * @prop --calcite-panel-background-color: Specifies the background color of the component.
+ * @prop --calcite-panel-header-background-color: Specifies the background color of the component's footer.
+ * @prop --calcite-panel-footer-background-color: Specifies the background color of the component's footer.
+
+ * @prop --calcite-panel-space: Specifies the padding of the component's `"unnamed (default)"` slot.
+ * @prop --calcite-panel-header-content-space: Specifies the padding of the `"header-content"` slot.
+ * @prop --calcite-panel-footer-space: Specifies the padding of the component's footer.
+
+ * @prop --calcite-panel-content-space: [Deprecated] Specifies the padding of the component's content. Use `--calcite-panel-space` instead.
```
Comment (alisonailea) [inline packages/calcite-components/src/components/panel/panel.scss:19]: Can you move the deprecated tokens to the bottom of the comment?

**Review by alisonailea — CHANGES_REQUESTED** [review]
Looks good! Just a couple of small things.

File: packages/calcite-components/src/components/panel/panel.scss (hunk 2 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/panel/panel.scss:16]: It's used here: https://github.com/Esri/calcite-design-system/pull/10822/files#diff-0b45120ffa61191421f1699207bb91da11d979e0a76a1b28d50353a3b17394c2R228
 
Screenshot 2024-11-21 at 10 48 49 AM

There's a separate example in the local demo and theme test for this "header-content-space" use case.

File: packages/calcite-components/src/components/panel/panel.scss (hunk 3 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/panel/panel.scss:21]: Swapped the order of these.

File: packages/calcite-components/src/components/panel/panel.scss (hunk 2 above)
Comment (alisonailea) [reply packages/calcite-components/src/components/panel/panel.scss:16]: missed that thanks!

File: packages/calcite-components/src/components/panel/panel.scss (hunk 1 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/panel/panel.scss:31]: Yeah I don't know if this component needs `@extend %component-host;` at all which places that visible background that isn't clipped. Might be able to remove that and not have to set the radius on :host

**Review by alisonailea — APPROVED** [review]
🚀

File: packages/calcite-components/src/components/panel/panel.scss (hunk 1 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/panel/panel.scss:31]: I was able to just take what was needed from `%component-host` and not set a background color on :host. Could you verify: https://www.chromatic.com/build?appId=6266d45509d7eb004aa254fb&number=6466

Comment (macandcheese) [reply packages/calcite-components/src/components/panel/panel.scss:31]: Going to merge in.

**Pull request merged**